app/admin/page.tsx

```diff
@@ -1,44 +1,66 @@
 import InstanceCard from "@/components/instance-card";
 import { EC2 } from "@aws-sdk/client-ec2";
 import { getEnvironmentPrefix } from "@/app/api/constants";
+import { retryWithBackoff } from "@/scripts/utils";
 
 async function getCurrentPRNumber(): Promise<string | null> {
   if (process.env.VERCEL_GIT_PULL_REQUEST_ID) {
+    console.log(`✅ Found PR from VERCEL_GIT_PULL_REQUEST_ID: ${process.env.VERCEL_GIT_PULL_REQUEST_ID}`);
     return process.env.VERCEL_GIT_PULL_REQUEST_ID;
   }
   
   const commitRef = process.env.VERCEL_GIT_COMMIT_REF;
   if (commitRef) {
     const branchName = commitRef.replace('refs/heads/', '');
+    
+    if (branchName.startsWith('devin/')) {
+      const prNumber = branchName.replace('devin/', '').split('-')[0];
+      if (prNumber && /^\d+$/.test(prNumber)) {
+        console.log(`✅ Found PR from branch name regex: ${prNumber}`);
+        return prNumber;
+      }
+    }
+    
     const githubToken = process.env.GITHUB_TOKEN;
     const githubRepo = process.env.GITHUB_REPOSITORY;
     
     if (githubToken && githubRepo && branchName) {
```

Comment:
Check all three of these individually, and raise an error if not defined, in a way that we'd see the error on the frontend